src/compiler/core.ts

```diff
@@ -1787,14 +1787,21 @@ namespace ts {
      *        (0.4 allows 1 substitution/transposition for every 5 characters,
      *         and 1 insertion/deletion at 3 characters)
      */
-    export function getSpellingSuggestion<T>(name: string, candidates: T[], getName: (candidate: T) => string | undefined): T | undefined {
+    export function getSpellingSuggestion<T>(name: string, candidates: T[], getName: (candidate: T) => string | undefined, isJsxAttr: boolean): T | undefined {
         const maximumLengthDifference = Math.min(2, Math.floor(name.length * 0.34));
         let bestDistance = Math.floor(name.length * 0.4) + 1; // If the best result isn't better than this, don't bother.
         let bestCandidate: T | undefined;
         let justCheckExactMatches = false;
         const nameLowerCase = name.toLowerCase();
         for (const candidate of candidates) {
             const candidateName = getName(candidate);
+            if (isJsxAttr) {

```

Comment:
Why not add these extra names in candidates list when using Jsx .. that seems better way to get correct suggestions by using existing algorithm rather than having to hard code these

Reply:
All attributes are in the candidates list including className and htmlFor. But they can't be selected out (therefore need hard code) by this algorithm because it think "class"  is "too far" to className.

Since they are really the special cases in react I think it's okay.